lib/graphql_metrics/extractor.rb

```diff
@@ -2,117 +2,76 @@
 
 module GraphQLMetrics
   class Extractor
-    CONTEXT_NAMESPACE = :extracted_metrics
-    TIMING_CACHE_KEY = :timing_cache
-    START_TIME_KEY = :query_start_time
+    class DummyInstrumentor
+      def after_query_start_and_end_time
+        [nil, nil]
+      end
+
+      def after_query_resolver_times(_ast_node)
+        []
+      end
+
+      def ctx_namespace
+        {}
+      end
+    end
 
     EXPLICIT_NULL = 'EXPLICIT_NULL'
     IMPLICIT_NULL = 'IMPLICIT_NULL'
     NON_NULL = 'NON_NULL'
 
-    attr_reader :query, :ctx_namespace
-
-    def self.use(schema_definition)
-      extractor = self.new
-      return unless extractor.extractor_defines_any_visitors?
-
-      extractor.setup_instrumentation(schema_definition)
-    end
-
-    def use(schema_definition)
-      return unless extractor_defines_any_visitors?
-      setup_instrumentation(schema_definition)
-    end
+    attr_reader :query
 
-    def setup_instrumentation(schema_definition)
-      schema_definition.instrument(:query, self)
-      schema_definition.instrument(:field, self)
+    def initialize(instrumentor = DummyInstrumentor.new)
+      @instrumentor = instrumentor
     end
 
-    def before_query(query)
-      return unless extractor_defines_any_visitors?
-
-      ns = query.context.namespace(CONTEXT_NAMESPACE)
-      ns[TIMING_CACHE_KEY] = {}
-      ns[START_TIME_KEY] = Process.clock_gettime(Process::CLOCK_MONOTONIC)
-    rescue StandardError => ex
-      handle_extraction_exception(ex)
+    def instrumentor
```

Comment:
This is here since I'd rather default again at runtime if an extractor calls `instrumentor`, and the extractor overrode `initialize` but didn't call `super`.